docs/design-proposals/multi-tenancy/multi-tenancy-network.md

```diff
@@ -120,10 +128,118 @@ The reasons that DNS service is per-network instead of per-space are:
+
+One solution is to making its name colliding space to tenant+namespace+network, which is also breaking and inconsistent with other types (e.g. introducing network as one more parameter to call kubectl get service).
+
+Another solution is to change their name to schema that can uniquely differentiate by combining with network name; this is awkward and breaking in general. 
+
+One plausible remedy is making stringent containment of network to namespaces - resources in one namespace belongs solely to one network; one network can accommodate multiple namespace. This cannot handle the default/kubernetes service problem, as it is of default namespace.
+
+None of the two of above seems ideal. However, the latter, fiddling with the service name, if coupled with trick applicable only to kubernetes & kube-dns, seems pragmatic and practical. The idea is suffixing service name with network name (e.g. kubernetes_mynetwork), also answering query for kubernetes with that of kubernetes_mynetwork (by alias record or other form of dynamic technique), so that pods are able to look up them without using the network-suffixed names. This approach is not flexible and depending on dns ability; it works fine with limited number of well-known aliases, though. coreDNS, one of the most commonly used dns server binaries, has such dynamic aliasing ability.
+
+One of the confusions is ```kubectl get service kubenetes``` would fail with no such resource error - this is breaking, but for good reason - to remind users of service resource being network specific in our system. However, pods are able to query DNS for kubernetes without caring about the new name schema as they are network related and able to locate the proper DNS service in the interesting network scope.
+
+#### Service IP
+
+Service IP address is allocated from the network-specific pool. 
+
+These are two types of service IP pool:
+
+##### Arktos managed pool
+Service IP addresses are explicitly managed by Arktos for every network. Each network might have different range of service IPs.
+
+Service IPAM is provided through 2 changes:
+
+1. each network object specifies its service IP range;
+
+    ```yaml
+    apiVersion: v1
+    kind: Network
+    metadata:
+      name: default
+    spec:
+      type: flat
+      service:
+      - cidr: 10.0.0.0/16
+      - cidr: 192.168.0.0/24
+    ```
+   
+1. on service creation, some mechanism to assign service the proper IP address.
+
+##### Network provider managed pool
+Service IP addresses are implicitly managed by the capable network provider. Network object can have the service IP cidr (or not); at best for information - Arktos delegates IPAM to the external network provider, via network controller invoking the service it provides. 
```

Comment:
In case of an external network provider, is there a specific mutation expected to the service object? For example, in Mizar, we would allocate an IP address and will need to mutate the service object with the IP value. Will Arktos permits mutation in this case or provides another mechanism to propagate this info?

Reply:
This is the detailed interaction spec between arktos controller and network provider. It is very important for the proper integration. What you described is very critical; we need to discuss more definitely - in separate meetings.

Whether the service ip address mutable or  not in arktos, I am not very clean - I'll find it out. If we can keep the service IP address inmutable, it would be ideal. 

I'll put a note in doc about this service mutation issue.